traitsui/qt4/helper.py

```diff
@@ -173,12 +173,20 @@ def __init__(self, icon, slot):
 
         # Configure the button.
         self.setIcon(ico)
-        self.setMaximumSize(ico_sz, ico_sz)
         self.setFlat(True)
         self.setFocusPolicy(QtCore.Qt.NoFocus)
 
         self.clicked.connect(slot)
 
+    def sizeHint(self):
+        # We want the button to have a size similar to the icon.
+        # Using the size computed for a tool button gives a desirable size.
+        option = QtGui.QStyleOptionButton()
+        self.initStyleOption(option)
+        size = self.style().sizeFromContents(
+            QtGui.QStyle.CT_ToolButton, option, option.iconSize
+        )
+        return size
```

Comment:
The added `sizeHint` method is covered by test: if I try to return anything else that is not an instance of QSize, the test suite would fail. However any assertion on the returned size is going to be tricky: it varies from system to system.